plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergSnapshots.java

```diff
@@ -13,27 +13,298 @@
  */
 package io.trino.plugin.iceberg;
 
+import io.trino.spi.QueryId;
 import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.MaterializedRow;
 import io.trino.testing.QueryRunner;
+import io.trino.testng.services.Flaky;
 import org.testng.annotations.Test;
 
+import java.util.ArrayList;
 import java.util.List;
 
-import static io.trino.plugin.iceberg.IcebergQueryRunner.createIcebergQueryRunner;
+import static io.trino.plugin.iceberg.IcebergMetadata.TRINO_QUERY_ID_NAME;
+import static io.trino.plugin.iceberg.IcebergQueryRunner.ICEBERG_CATALOG;
 import static io.trino.testing.sql.TestTable.randomTableSuffix;
 import static java.lang.String.format;
 import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.Assertions.assertThat;
 
 public class TestIcebergSnapshots
         extends AbstractTestQueryFramework
 {
     private static final int ID_FIELD = 0;
 
+    static final String ADD_QUERY_ID_TO_ICEBERG_SNAPSHOT_SUMMARY_ISSUE = "https://github.com/trinodb/trino/issues/11292";
+
     @Override
     protected QueryRunner createQueryRunner()
             throws Exception
     {
-        return createIcebergQueryRunner();
+        return IcebergQueryRunner
+                .builder()
+                .addIcebergProperty("iceberg.expire_snapshots.min-retention", "0s")
+                .build();
+    }
```

Comment:
If we don't include the tests to verify that expiring snapshots does not produce a new snapshot, this change can be reverted.